Source/Core/VideoCommon/TextureConverterShaderGen.cpp

```diff
@@ -152,133 +180,73 @@ ShaderCode GenerateTextureConverterShaderCode(APIType api_type,
     switch (uid_data->dst_format)
     {
     case EFBCopyFormat::R4:  // R4
-      colmat[0] = colmat[4] = colmat[8] = colmat[12] = 1;
```

Comment:
What do you think about using:
if (!uid_data->efb_has_alpha) out.Write("texcol.a = 1.0;\n");
before this switch?